### ISSUE TITLE
Dead or Alive 2 -  Glitched graphics effects, sound effect and UI transition (non-OIT)

### ISSUE THREAD

**Issue by PixelCrusader** [libretro/flycast#181]
Windows 7 64 bit, AMD Radeon HD 6450, Intel i5-2320 CPU @ 3.00GHz
RetroArch 1.7.3 - Reicast 0.1 83c8c07

- The fireflies and lamps of the nighttime Aerial Garden stage aren't displaying properly.

![dead or alive 2 usa -180713-182218](https://user-images.githubusercontent.com/40935002/42729118-ce791fb8-879b-11e8-88c2-c36b2241423a.png)
![dead or alive 2 usa -180713-182234](https://user-images.githubusercontent.com/40935002/42729120-d2d79738-879b-11e8-8496-eb9e67888272.png)
![dead or alive 2 usa -180713-182325](https://user-images.githubusercontent.com/40935002/42729128-f7309ab2-879b-11e8-82f8-00a99f8e3ef3.png)
![dead or alive 2 usa -180713-182447](https://user-images.githubusercontent.com/40935002/42729129-03c83898-879c-11e8-8f68-22fbeeb98809.png)

- There is a sound bug during a cutscene in this stage where the sound of running water is cut off prematurely.

Here are three methods to get to the stage:
1. Options > Game Settings > Versus Mode > Vs Stage Selector. Choose anything but off. Exit until you're back at the main menu and choose Versus. 

Pick your characters. You'll need an active second controller to proceed, so either configure another controller or bind controller 2's inputs to the gamepad you're already using. RetroArch > Settings > Input > User 2 Binds > User 2 Device Index.

Use the right trigger to select The Aerial Garden (down and to the left) other buttons will send you to the daytime version of the stage.

2. Select Jahn Lee (top, second from right) in Story Mode and win three matches, which are only a single round. (You need to do this in order to play the cutscene in which the sound bug happens.)

3. Do nothing at the start screen and let the game go into attract mode. The affected stage is included in the sequence.

Reference video: [1](https://www.youtube.com/watch?v=ZWMzP0wcQBo#t=1m46s) [2](https://www.youtube.com/watch?annotation_id=annotation_739230325&feature=iv&src_vid=QM7lytqOmkY&v=3RLdIVmZ9cw#t=5m40s) 

- There is a minor visual bug that occurs when going from the start screen to the main menu. The screen turns gray for a moment before returning to normal. Unable to capture a picture of it.

Reference video: [1](https://www.youtube.com/watch?annotation_id=annotation_3684812813&feature=iv&src_vid=S0tp0_GkqjE&v=aqpZMbJF5Kk#t=3m38s) [2](https://www.youtube.com/watch?annotation_id=annotation_739230325&feature=iv&src_vid=QM7lytqOmkY&v=3RLdIVmZ9cw#t=3m35s)

**Comment by inactive123** [comment]
Regarding the first issue - I don't get this at all - these are some screenshots taken with Reicast OIT -

![image](https://user-images.githubusercontent.com/1075274/42731688-f7a6852a-8812-11e8-846b-598bfd0cca1d.png)

![image](https://user-images.githubusercontent.com/1075274/42731690-01d1a908-8813-11e8-82da-55929a46c263.png)

![image](https://user-images.githubusercontent.com/1075274/42731692-061230a0-8813-11e8-971d-950885b91996.png)

**Comment by inactive123** [comment]
OK, I see it now - it happens with the non-OIT renderer. Reicast OIT renders it properly.

![image](https://user-images.githubusercontent.com/1075274/42731713-9e9cb2a0-8813-11e8-8e93-67abfefb6995.png)

@flyinghead Is this a regression or was it always like this with the non-OIT renderer?

**Comment by flyinghead** [comment]
The shading instruction for mode DECAL is missing in the fragment shader:
```
diff --git a/core/rend/gles/gles.cpp b/core/rend/gles/gles.cpp
index 518f5e8..f6f9a2a 100644
--- a/core/rend/gles/gles.cpp
+++ b/core/rend/gles/gles.cpp
@@ -225,6 +225,11 @@ void main() \n\
                                if (cp_AlphaTestValue>texcol.a) discard;\n\
                        #endif  \n\
                #endif \n\
+               #if pp_ShadInstr==0 \n\
+               { \n\
+                       color=texcol; \n\
+               } \n\
+               #endif\n\
                #if pp_ShadInstr==1 \n\
                { \n\
                        color.rgb*=texcol.rgb; \n\
```

**Comment by inactive123** [comment]
Thanks, that fixes it! Committed it.

**Comment by inactive123** [comment]
I notice the sound bug @PixelCrusader. I think the sound bug is quite similar to the water drop effect in Resident Evil Code: Veronica (beginning of the game, after walking through the cemetary). It gets weirdly cut off at points and loops awkwardly. Seems to be similar to this sound effect being abruptly cut off and then looped.

**Comment by PixelCrusader** [comment]
I'm confirming that the graphical glitches have been resolved on Non-OIT Reicast 0.1 2d411a1, if not earlier.
